File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> pair;
        unordered_map<int, int> map;
        
        for (int i = 0; i < nums.size(); i++) {
            
            int elem1 = nums[i];
            auto pt2 = map.find(target - elem1);
            
            if (pt2 != map.end() ) {
                pair.emplace_back(i);
                pair.emplace_back(pt2 -> second);
                break;
            }
            
            map[nums[i]] = i;
        }
        return pair;
    }
};

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map = {}
    
    for (let i = 0; i < nums.length; i++) {
        
        let elem1 = nums[i];
        let pt2 = map[target - elem1];
        
        if (pt2 != undefined) {
            return [i, pt2];
        }
        else {
            map[elem1] = i;
        }
        
    }
};